### ISSUE TITLE
Not able to identify Employees with multiple jobs in the Shared with list

### ISSUE THREAD

**Issue by steffipravasi** [bcgov/performance#1104]
Solution: Display The dept id with the format xxx-xxxx

Employee list:
181150	Lenarduzzi,Sophia	Sophia.Lenarduzzi@gov.bc.ca
178226	Diaz,Maria Delia	Maria.DeliaDiaz@gov.bc.ca
177102	Nadan,Devisha	Devisha.Nadan@gov.bc.ca
173226	Karas,Ellie	Ellie.Karas@gov.bc.ca
159617	Dureault,Bronte	Bronte.Dureault@gov.bc.ca
71231	Dashwood,Barbara Anne	bdashwood@islandstrust.bc.ca

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdkZFIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7288cb842dd979afd5b9e9006c3780d00cdeca2a/Screenshot%202023-08-28%20at%201.40.42%20PM.png)

Issue must be addressed at all the places:
- My Goals -> My current goals -> Shared with
- My team -> Team members -> Shared column
- HR/Sys admin -> Share Employees -> Shared modal

**Comment by Travis-A-Clark** [comment]
@steffipravasi Following our chat at scrum this morning, I don't know that we need to add in the Dept ID to this list. I don't think it will matter which one they choose since the content of the employee profile is the same no matter which job they select. Perhaps, instead, we can add a condition to remove duplicates from the drop down list so that the name and email combination is only shown once.

**Comment by steffipravasi** [comment]
@Travis-A-Clark But if we don't add the Dept id then the goal will be displayed in all the jobs. I thought that only the goals list related to the selected job must display and goals in the other job must be hidden.

**Comment by jizhaogit** [comment]
We are using "**user_id**" for the goal share module. so even a user has multi jobs with same employee id, we will still use his different **user_id** to do the sharing.  
that means, if we group the same employee id as one for removing duplicates, we will face a problem: which job we want the goal be shared with? (Which is Steffi worries if I not misunderstand).

**Comment by Travis-A-Clark** [comment]
Ah, okay. Didn't realize about the different user_id for the same employee id. I guess we do need this ticket still. Thanks!

**Comment by steffipravasi** [comment]
Noticing one issue which I am not sure about
Since we have some duplicate records, we have users with similar Dept-Id as well now
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBajk5IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--1a6b477f3c775f92c7f5bc9b6c87574f080638a4/Screenshot%202023-09-08%20at%201.21.58%20AM.png)

Note:

**Comment by jizhaogit** [comment]
i checked the user in the database and found that there are 2 similar record of this user in the employee table. Details can be found from the attachment

[employee_maria_diaz.xlsx](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBc2QvIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--910fbf8449edb9597cfec1341f2d56696e464f4d/employee_maria_diaz.xlsx)

the difference about the 2 records is the second record has no supervisor information. Don't know if that record is not in used anymore.

**Comment by steffipravasi** [comment]
The purpose of the ticket is achieved. Moving to PO review.
@Travis-A-Clark Do we want to create a separate ticket to handle the duplicate records issue which requires a data cleanup?

**Comment by Travis-A-Clark** [comment]
@telusdcinco will check the ODS data to see if we are still receiving two rows and then come up with a fix.

**Comment by Travis-A-Clark** [comment]
Creating new ticket to follow up on 178226 Diaz,Maria Delia

**Comment by Travis-A-Clark** [comment]
@steffipravasi @jizhaogit With all of the work on ticket #857 it has become apparent that we cannot actually share a goal just to one job profile for an employee...it goes on the same profile no matter which job is selected from share dropdown list. This wasn't clear when we started this ticket.

As such, I recommend going back and removing the Dept ID from the drop down list so that the entries are shorter (like they were before with just email) and instead adding a filter to only display each name / email combination once in the dropdown list. I believe this was an option Ji mentioned back when we started this ticket, and given the other realities of #857, this looks like the best option. Sorry to reverse course!

@steffipravasi Any concernse with this?
